src/python/pants/engine/legacy/graph.py

```diff
@@ -619,6 +620,46 @@ class HydratedField:
     return HydratedTarget(adaptor=type(target_adaptor)(**kwargs))
 
 
+@rule
+async def resolve_target(
+    hydrated_struct: HydratedStruct, registered_target_types: RegisteredTargetTypes
+) -> WrappedTarget:
```

Comment:
Nothing new here, other than being a rule. This used to be a pure function in `target.py`.

Reply:
Is it becoming a rule because it is expected to start awaiting things?

Reply:
It's becoming a rule so that in the followup PR we can start requesting `Targets` instead of `HydratedTargets`. Here's the rule to get `Targets`:

```python
@rule
async def resolve_targets(addresses: Addresses) -> Targets:
    wrapped_targets = await MultiGet(Get[WrappedTarget](Address, a) for a in addresses)
    return Targets(wrapped_target.target for wrapped_target in wrapped_targets)
```

It's also becoming a rule so that we can access the singleton `RegisteredTargetTypes` through dependency injection, rather than having to directly pass it as a parameter.